numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -175,6 +175,12 @@ string_multiply(Buffer<enc> buf1, npy_int64 reps, Buffer<enc> out)
         return;
     }
 
+    size_t width = out.buffer_width();
+    size_t pad = 0;
+    if (width < len1*reps) {
```

Comment:
```suggestion
    if (reps > NPY_MAX_SSIZE_T  // ssize_t is enough here
        || npy_mul_sizes_with_overflow(len1, reps, &width)
        || width > out.buffer_width()) {
            width = out.buffer_width;
```
Admittedly this is tedious and pedantic, but maybe we should do overflow checks?  `npy_mul_sizes_with_overflow` uses `ssize_t` inputs, which is safe here but may require casting to tell the compiler about that.  (Or just using the `size_t` version.)
I suppose `NPY_UNLIKELY()` will be irrelevant here.